### PR TITLE
Feat validation

### PR DESCRIPTION
**Pull request by SpliiT**

### CODE REVIEW

File: components/CrsSelector.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "$id": "/workflows/crs_selection",
```
Comment (JulienChampagnol) [inline components/CrsSelector.json:2]: Are you shure about the path ?

File: components/CrsSelector.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      }
+    },
+    "selected_crs": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    }
+  },
+  "required": ["search", "data_table_loading", "crs_list", "selected_crs"]
```
Comment (JulienChampagnol) [inline components/CrsSelector.json:37]: Required parameters are only those send to the API

File: components/FileSelector.vue [hunk 3]
```diff
@@ -15,15 +15,16 @@
 
 <script setup>
   const stepper_tree = inject("stepper_tree")
```
Comment (JulienChampagnol) [inline components/FileSelector.vue:17]: You can remove this line if you're only using it

File: components/Header.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+  "$id": "/workflows/tool_cards",
```
Comment (JulienChampagnol) [inline components/Header.json:2]: I don't think this component is making any API call?

File: components/Launcher.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+  "$id": "/workflows/captcha_validation",
```
Comment (JulienChampagnol) [inline components/Launcher.json:2]: Same here

File: components/Launcher.vue [hunk 6]
```diff
@@ -8,26 +8,61 @@
...
   const cloud_store = use_cloud_store()
-  const { is_captcha_validated } = storeToRefs(cloud_store)
+  const { is_cloud_running, is_captcha_validated, is_connexion_launched } =
+    storeToRefs(cloud_store)
+
+  const site_key = useRuntimeConfig().public.RECAPTCHA_SITE_KEY
 
   watch(is_captcha_validated, async (value) => {
     if (value === true && process.client) {
+      console.log("is_captcha_validated 2", is_captcha_validated.value)
```
Comment (JulienChampagnol) [inline components/Launcher.vue:39]: Remove all console.logs

File: components/PackagesVersions.vue [hunk 7]
```diff
@@ -19,41 +19,46 @@
...
+    console.log("watch", value)
     if (value === true) {
       get_packages_versions()
     }
   })
 
   onMounted(() => {
-    if (cloud_store.is_running === true) {
+    console.log("mounted", is_running)
+    if (is_running.value === true) {
```
Comment (JulienChampagnol) [inline components/PackagesVersions.vue:53]: I think you can do directly
```suggestion
    if (is_running.value) {
```

File: components/Step.vue [hunk 8]
```diff
@@ -38,6 +38,7 @@
     </v-row>
     <Transition name="slide-fade">
       <v-col v-if="step_index == current_step_index">
+        {{ steps[step_index].component.component_options }}
```
Comment (JulienChampagnol) [inline components/Step.vue:41]: To remove

File: composables/api_fetch.js [hunk 9]
```diff
@@ -1,21 +1,49 @@
...
     onRequestError({ error }) {
+      console.log("onRequestError", error)
       geode_store.stop_request()
-      // MUST STAY HERE FOR EASIER DEBUG
-      // console.log('onRequestError', error)
+      // Log the error to a proper logging library
       errors_store.add_error({
-        code: "",
+        code: error.code,
         route: request_url,
```
Comment (JulienChampagnol) [inline composables/api_fetch.js:47]:
```suggestion
        route: schema.$id,
```

File: stores/cloud.js [hunk 10]
```diff
@@ -1,3 +1,5 @@
+import { defineStore } from "pinia"
```
Comment (JulienChampagnol) [inline stores/cloud.js:1]: To remove, it's supposed to work with auto imports

File: components/Launcher.vue [hunk 11]
```diff
@@ -8,26 +8,61 @@
         align="center"
       >
         <h4 class="pb-3">Please complete the recaptcha to launch the app</h4>
-        <Recaptcha />
+        <vue-recaptcha
```
Comment (JulienChampagnol) [inline components/Launcher.vue:11]: I think they're has been some new changes here, keep the ones that are on the next branch

File: composables/api_fetch.js [hunk 12]
```diff
@@ -25,19 +53,18 @@ export function api_fetch(
...
       }
     },
     onResponseError({ response }) {
+      console.log(response)
       geode_store.stop_request()
-      // MUST STAY HERE FOR EASIER DEBUG
-      // console.log('onResponseError', response)
       errors_store.add_error({
         code: response.status,
         route: request_url,
```
Comment (JulienChampagnol) [inline composables/api_fetch.js:70]: Same here

File: package.json
Comment (JulienChampagnol) [inline package.json:1]: Think about upgrading your dependencies, but check if they work

**Review by JulienChampagnol — CHANGES_REQUESTED** [review]

File: components/Launcher.vue (hunk 6 above)
Comment (SpliiT) [reply components/Launcher.vue:39]: I'll do it once all the errors are resolved

File: components/Launcher.vue [hunk 13]
```diff
@@ -13,56 +13,27 @@
...
-    if (process.client) {
-      const config = useRuntimeConfig()
-      if (config.public.NODE_ENV !== "production") {
-        cloud_store.$patch({ is_captcha_validated: true })
-        console.log("is_captcha_validated", is_captcha_validated.value)
-      }
-    }
-  })
-
-  async function submit_recaptcha(token) {
```
Comment (BotellaA) [inline components/Launcher.vue]: Why this deletion?

**Review by BotellaA — APPROVED** [review]

**Pull request merged**